File: www/html/server_demo/picmd.php
<?php
$action = isset($_POST["action"]) ? $_POST["action"] : null;
$cmd = isset($_POST["cmd"]) ? $_POST["cmd"] : null;

if ($action === "set-linux-cmd" && !empty($cmd)) {
    $filePath = "/home/levi/Pi_Controlcmd/picmd.txt";
    
    $myfile = fopen($filePath, "a");
    if ($myfile) {
        fwrite($myfile, $cmd);
        fclose($myfile);
        
        $str = file_get_contents($filePath);
        echo $str;
    } else {
        echo "Unable to open file!";
    }
} else {
    echo "Invalid request!";
}
?>

File: Pi_Controlcmd/picmd.py
import os
import time

while True:
    with open("/home/levi/Pi_Controlcmd/picmd.txt", "r") as file1:
        picmd = file1.read().strip() 

    if picmd:
        print(f"read command: {picmd}")  
        with open("/home/levi/Pi_Controlcmd/picmd.txt", "w") as file:
            file.write("") 
        print(f"operate command: {picmd}") 
        os.system(picmd + ' &') 

    time.sleep(1)  
